packages/eslint-plugin/src/util/explicitReturnTypeUtils.ts

```diff
@@ -264,6 +265,26 @@ function checkFunctionReturnType(
   report(getReporLoc(node, sourceCode));
 }
 
+function isTypedFunctionExpression(
+  node: TSESTree.ArrowFunctionExpression | TSESTree.FunctionExpression,
+  options: Options,
+): boolean {
+  const parent = nullThrows(node.parent, NullThrowsReasons.MissingParent);
```

Comment:
small question about `nullThrows`, is there a reason that we have to return parent instead of using asserts?

```diff
- function nullThrows<T>(value: T | null | undefined, message: string): T {
+ function nullThrows(value: unknown, message: string): asserts value {
```

this is off topic, i'm just curious